Mage.Sets/src/mage/cards/a/AnimalFriend.java

```diff
@@ -0,0 +1,147 @@
+import mage.abilities.dynamicvalue.common.EquipmentAttachedCount;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.abilities.effects.common.continuous.GainAbilityWithAttachmentEffect;
+import mage.constants.*;
+import mage.counters.CounterType;
+import mage.filter.common.FilterControlledPermanent;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.SquirrelToken;
+import mage.target.common.TargetCreaturePermanent;
+import mage.abilities.effects.common.AttachEffect;
+import mage.target.TargetPermanent;
+import mage.abilities.keyword.EnchantAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+
+/**
+ *
+ * @author grimreap124
+ */
+public final class AnimalFriend extends CardImpl {
+
+
+
+    public AnimalFriend(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{1}{G}");
+        
+        this.subtype.add(SubType.AURA);
+
+        // Enchant creature
+        TargetPermanent auraTarget = new TargetCreaturePermanent();
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.BoostCreature));
+        this.addAbility(new EnchantAbility(auraTarget));
+
+        // Enchanted creature has "Whenever this creature attacks, create a 1/1 green Squirrel creature token. Put a +1/+1 counter on that token for each Aura and Equipment attached to this creature other than Animal Friend."
+        this.addAbility(new SimpleStaticAbility(new AnimalFriendEffect()));
+
+    }
+
+    private AnimalFriend(final AnimalFriend card) {
+        super(card);
+    }
+
+    @Override
+    public AnimalFriend copy() {
+        return new AnimalFriend(this);
+    }
+}
+
+class AnimalFriendEffect extends GainAbilityWithAttachmentEffect {
+    private static final FilterControlledPermanent filter = new FilterControlledPermanent(SubType.AURA);
+
+    static {
+        filter.add(AnotherPredicate.instance);
+    }
+
+    AnimalFriendEffect() {
+        super("Enchanted creature has \"Whenever this creature attacks, create a 1/1 green Squirrel creature token. " +
+                        "Put a +1/+1 counter on that token for each Aura and Equipment attached to this creature other than Animal Friend.\"",
+                (Effect) null, null, null);
+    }
+
+    private AnimalFriendEffect(final AnimalFriendEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public AnimalFriendEffect copy() {
+        return new AnimalFriendEffect(this);
+    }
+
+    @Override
+    protected Ability makeAbility(Game game, Ability source) {
+        if (source == null || game == null) {
+            return null;
+        }
+        return new AttacksTriggeredAbility(new AnimalFriendTokenEffect(source.getSourceId()), false, "");
```

Comment:
so you're going to want to pass a MageObjectReference not just the source id. Also you should pass in the name of the source permanent for use in the text